src/EditDistPeriodicityDetector.cpp

```diff
@@ -45,70 +46,94 @@ namespace geopm
 {
     EditDistPeriodicityDetector::EditDistPeriodicityDetector(int history_buffer_size)
         : m_history_buffer(history_buffer_size)
+        , m_history_buffer_size(history_buffer_size)
         , m_period(-1)
         , m_score(-1)
+        , m_record_count(0)
     {
-
+        m_DP = geopm::make_unique<std::vector<uint32_t> >(history_buffer_size * history_buffer_size * history_buffer_size);
```

Comment:
Now that I see the context for this, I think this can be declared on the stack with
```std::vector<uint32_t> m_DP;``` in the header, and initialized in the list above with ```m_DP(history_buffer_size * history_buffer_size * history_buffer_size)```.  Then you don't need to dereference below